storage/modules/block_storage.go

```diff
@@ -1084,6 +1136,96 @@ func (b *BlockStorage) FindTransaction(
 	return newestBlock, newestTransaction, nil
 }
 
+func (b *BlockStorage) FindRelatedTransactions(
+	ctx context.Context,
+	transactionIdentifier *types.TransactionIdentifier,
+	db database.Transaction,
+) (*types.BlockIdentifier, *types.Transaction, []*types.TransactionIdentifier, error) {
```

Comment:
I'm thinking it makes more sense to return `(*types.BlockIdentifier, *types.Transaction, []*types.Transaction, error)` so the caller doesn't need to re-fetch all related transactions. This won't incur any additional performance overhead for `FindRelatedTransactions` because each call to `FindTransaction` already fetches the `*types.Transaction`.